runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -208,29 +208,24 @@ NNFW_STATUS nnfw_session::create(nnfw_session **session)
 {
   if (session == nullptr)
     return NNFW_STATUS_UNEXPECTED_NULL;
-
-  // Create session
-  *session = new (std::nothrow) nnfw_session();
-  if (*session == nullptr)
+  try
   {
-    std::cerr << "Error during session creation" << std::endl;
-    return NNFW_STATUS_OUT_OF_MEMORY;
+    auto new_session = std::unique_ptr<nnfw_session>(new nnfw_session());
+    new_session->_kernel_registry = std::make_shared<onert::api::CustomKernelRegistry>();
+    *session = new_session.release();
   }
-
-  // Initialize fields
-  try
+  catch (const std::bad_alloc &e)
   {
-    (*session)->_kernel_registry = std::make_shared<onert::api::CustomKernelRegistry>();
+    std::cerr << "Error during session creation" << std::endl;
+    *session = nullptr; // Set nullptr on error to keep the old behavior
+    return NNFW_STATUS_OUT_OF_MEMORY;
   }
   catch (const std::exception &e)
   {
     std::cerr << "Error during session initialization : " << e.what() << std::endl;
-    delete *session;
-    *session = nullptr;
-
+    *session = nullptr; // Set nullptr on error to keep the old behavior
```

Comment:
I have two questions.
1. Is there any guarantee that the old behavior was `nullptr`? If so, it is necessary to add a code checking `nullptr` before allocating.
2. Is there any case where `*session` was assigned in this block?

And There is a case where `new_session` is not deleted if not called `new_session.release()`. It's memory leak.

Reply:
3. `session` is out variable (pointer to pointer to session). 

> I have two questions.
> 
> 1. Is there any guarantee that the old behavior was `nullptr`?

The old implementations set `session` to `nullptr` in case of out-of-memory error. New code also set to `nullptr` in case of out-of-memory error to keep this old behavior.

> If so, it is necessary to add a code checking `nullptr` before allocating.

I don't understand what you mean by checking `nullptr`. Do you mean the 1st null check for output variable's address? Or do you mean the 2nd null check?

> 2. Is there any case where `*session` was assigned in this block?

I don't understand what you mean by this block. It sets the passed value in `*session`. It may be have some value prior to `nnfw_session::create.

> And there is a case where `new_session` is not deleted if not called `new_session.release()`. 

Again, I don't understand what you mean. `nnfw_session.release()` is to get the ownership of `nnfw_session` allocation.

> It's memory leak.

Could you explain more? From `session` or `_kernel_registry`? When does it happen?

Reply:
> The old implementation sets session (output parameter) to nullptr in case of out-of-memory. The content in *session will be overwritten to nullptr. New code also sets to nullptr in case of out-of-memory error to keep this old behavior.

The old implementation also overwrites `*session` to `nullptr`. ㄲegardless of the fact, do you mean the old behavior is that `*session ` was `nullptr` before called this function? If so, I thought it would be better to add an exception handling with checking `*session != nullptr`. 

> I don't understand what you mean by checking nullptr. Do you mean the 1st null check for output variable's address? Or do you mean the 2nd null check?

I mean the `*session` null check.

> Could you please explain more? From session or _kernel_registry? When does it happen?

Sorry, I missed `unique_ptr`. Please ignore about memory leak.

> I don't understand what you mean by this block. It sets the passed value in *session. It may be have some value prior to nnfw_session::create.

My question was related to memory leak. Please ignore my 2nd question.

Reply:
> The old implementation also overwrites `*session` to `nullptr`. 

Yes, it is exactly what I mean by "to keep the old behavior". I don't change the old behavior.

> Regardless of the fact, do you mean the old behavior is that `*session ` was `nullptr` before called this function? 

No. Old behavior what I mean is setting `*session` to `nullptr` on out-of-memory.